src/hb-ot-layout.cc

```diff
@@ -1046,6 +1064,93 @@ hb_ot_layout_get_size_params (hb_face_t    *face,
   return false;
 }
 
+/**
+ * hb_ot_layout_get_feature_params:
+ * @face: #hb_face_t to work upon
+ * @tag: ssXX and cvXX tag
+ * @label_id: (out) (allow-none): The ‘name’ table name ID that specifies a string
+ *            for a user-interface label for this feature. (May be NULL.)
+ * @tooltip_id: (out) (allow-none): The ‘name’ table name ID that specifies a string
+ *              that an application can use for tooltip text for this
+ *              feature. (May be NULL.)
+ * @sample_id: (out) (allow-none): The ‘name’ table name ID that specifies sample text
+ *             that illustrates the effect of this feature. (May be NULL.)
+ * @num_named_parameters: (out) (allow-none):  Number of named parameters. (May be zero.)
+ * @first_param_id: (out) (allow-none): The first ‘name’ table name ID used to specify
+ *                  strings for user-interface labels for the feature
+ *                  parameters. (Must be zero if numParameters is zero.)
+ * @char_count: (in/out) (allow-none): The count of characters for which this feature
+ *              provides glyph variants. (May be zero.)
+ * @characters: (out) (allow-none): A buffer pointer. The Unicode Scalar Value
+ *              of the characters for which this feature provides glyph variants.
+ *
+ * Returns bunch of NameIDs and other related information for a tag,
+ * and sets NameIDs parameters (suffixed with _id) to 0 in case
+ * the requested tag not found or the tag doesn't support it.
+ *
+ * Return value: true if could find any feature with the tag, false otherwise
+ *
+ * Since: REPLACEME
+ **/
+hb_bool_t
+hb_ot_layout_get_feature_params (hb_face_t      *face,
+				 hb_tag_t        tag,
+				 unsigned int   *label_id, /* OUT */
+				 unsigned int   *tooltip_id, /* OUT */
+				 unsigned int   *sample_id, /* OUT */
+				 unsigned int   *num_named_parameters, /* OUT */
+				 unsigned int   *first_param_id, /* OUT */
+				 unsigned int   *char_count, /* IN/OUT */
+				 hb_codepoint_t *characters /* OUT */)
+{
+  const OT::FeatureParams &feature_params =
+    _get_gsubgpos_matched_feature_params (face, tag);
+  if (&feature_params != &OT::Null (OT::FeatureParams))
+  {
+    const OT::FeatureParamsStylisticSet& stylistic_set_params =
+      feature_params.get_stylistic_set_params (tag);
+    if (&stylistic_set_params != &OT::Null (OT::FeatureParamsStylisticSet)) /* ssXX */
+    {
+#define PARAM(a, A) if (a) *a = A
+      PARAM(label_id, stylistic_set_params.uiNameID);
+      // ssXX features don't have the rest
+      PARAM(tooltip_id, 0);
+      PARAM(sample_id, 0);
+      PARAM(num_named_parameters, 0);
+      PARAM(first_param_id, 0);
+      PARAM(char_count, 0);
+      return true;
+    }
+    const OT::FeatureParamsCharacterVariants& character_variants_params =
```

Comment:
Small nit; I’d personaly prefer shorter variable names for `stylistic_set_params` and `character_variants_params` variables, may be `ss_params` and `cv_params`.